lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -3754,11 +3757,6 @@ ExpandMacroExpansionDeclRequest::evaluate(Evaluator &evaluator,
   if (!expandFreestandingDeclarationMacro(MED, expandedTemporary))
     return {};
   auto expanded = ctx.AllocateCopy(expandedTemporary);
-  // FIXME: Handle this in name lookup instead of `addMember`.
-  // MED->setRewritten(expanded);
-  if (auto *parentDecl = MED->getDeclContext()->getAsDecl())
-    if (auto *idc = dyn_cast<IterableDeclContext>(parentDecl))
-      for (auto *decl : expanded)
-        idc->addMember(decl);
+  MED->setRewritten(expanded);
```

Comment:
It's not critical, but a more idiomatic way to do this would be to make `ExpandMacroExpansionDeclRequest` cached and drop the "rewritten" field from the structure.

Reply:
Will do that in a follow-up!

Reply:
It's already cached in the request. Just need to remove the field.

Reply:
https://github.com/apple/swift/pull/63486